controller/PlatformAdmin.php

```diff
@@ -34,6 +37,18 @@
  */
 class PlatformAdmin extends tao_actions_SaSModule
 {
+    public function initialize()
+    {
+        parent::initialize();
+
+        /** @var EventManager $eventManager */
+        $eventManager = $this->getServiceManager()->get(EventManager::SERVICE_ID);
+        $eventManager->attach(
+            ResourceCreated::class,
+            [UpdatePlatformRegistrationSnapshotListener::SERVICE_ID, 'whenResourceCreated']
```

Comment:
This is clever!

But IMO it can cause some unwanted side effects. Like in case the controller would _indirectly_ call one of the deeply nested services that would happen to also emit `ResourceCreated` event.

Wouldn't it be safer to actually subscribe to that event, and filter out the unwanted types of resources in `whenResourceCreated`?

Reply:
Or keep the `attach()` here, but add some filtering to `whenResourceCreated()` anyway.

Reply:
good idea!

Reply:
https://github.com/oat-sa/extension-tao-lti/pull/339/commits/d9c62c55404b49e08f0f173c104fd09c5a6442ae